modAionImpl/src/org/aion/zero/impl/db/AionRepositoryImpl.java

```diff
@@ -677,6 +686,16 @@ public void close() {
                 LOGGEN.error("Exception occurred while closing the block store.", e);
             }
 
+            try {
+                if (pendingStore != null) {
+                    pendingStore.close();
+                    LOGGEN.info("Pending block store closed.");
+                    pendingStore = null;
+                }
+            } catch (Exception e) {
+                LOGGEN.error("Exception occurred while closing the pending block store.", e);
```

Comment:
The close method is called only at shutdown, so the kernel will not continue running. Data may get corrupted if the databases are not closed correctly, but this is not a big issue in the case of the pending store since any blocks that cannot be loaded from storage can be requested by the sync protocol.